### PR TITLE
Windows syscall support

### PR DESCRIPTION
**Pull request by yakman2020**
Adds posix syscalls to windows host. 

fcntl flocking and ioctls are not supported.

### CODE REVIEW

**Comment by yakman2020** [comment]
bors try

**Comment by anitagov** [comment]
Wondering if you can rebase your branch to master and resolve the conflicts.

**Comment by jxyang** [comment]
General thoughts: 
* I think it's a good idea to break up the file like Mike did for Linux syscalls. 
* It would be nice we document in the code which APIs we have achieved behavior parity with their Linux counterpart, and which APIs are still work in progress

File: host/windows/syscall.c [hunk 1]
```diff
@@ -30,334 +30,2158 @@
...
+
+/*
+**==============================================================================
+**
+** Local definitions.
+**
+**==============================================================================
+*/
+
+static BOOL _winsock_inited = 0;
```
Comment (anitagov) [inline host/windows/syscall.c]: Consider changing to _winsock_initialized

**Comment by jhand2** [comment]
Closing in favor of #2342

**Pull request closed without merging**